scripts/get-prs

```diff
@@ -20,13 +36,93 @@ import sys
 
 from github import Github
 
-test_start_re = "[Tt]est with:?"
-repo_ref_re = "([a-zA-Z0-9_-]+/[a-zA-Z0-9_-]+#[0-9]+)"
-pr_re = re.compile(repo_ref_re)
-repo_sep_re = "([, ]+(and)?[ ]*)"
-test_re = re.compile(f"{test_start_re}[ ]*({repo_ref_re}{repo_sep_re}?)+")
+# Start with a few simple regex'es and re-use them to build more complex ones.
+# Using "(?:...)" makes a regex a non-capturing group, which is basically an
+# atomic expression again that operators can be applied to.
+def regex_atom(regex):
+    return f"(?:{regex})"
 
-if __name__ == '__main__':
+whitespace_re = regex_atom("[ \t]")
+org_re        = regex_atom("[a-zA-Z0-9_-]+")
+repo_re       = regex_atom("[a-zA-Z0-9_-]+")
+pr_id_re      = regex_atom("[0-9]+")
+
+# Now things get a bit tricky, Both "pr_ref_short_re" and "pr_ref_url_re" are
+# atoms, but they have matching group inside. When or'ing multiple regex'es,
+# each sub-regex contribute the groups, even if they did not match. Thus, any
+# match of "pr_ref" will yield a 3-tupel, where either the first element or the
+# last two elements ("org_re/repo_re" and "pr_id_re") are not empty. That nicely
+# allows determining what PR reference form we have found.
+pr_ref_short_re = regex_atom(f"({org_re}/{repo_re}#{pr_id_re})")
+pr_ref_url_re = regex_atom(f"https://github.com/({org_re}/{repo_re})/pull/({pr_id_re})")
+pr_ref = regex_atom(f"{pr_ref_short_re}|{pr_ref_url_re}")
+
+# Since a test can run with PRs from multiple repos, we have to support lists
+# also. A hacky way to achieve this is just defining an atom that is a list
+# item with optionally another separator-atom attached - so a list is just a
+# sequence of such items. That is good enough practically.
+sep_comma_re    = regex_atom(f"{whitespace_re}*,{whitespace_re}*")
+sep_and_re      = regex_atom(f"{whitespace_re}+and{whitespace_re}+")
+sep_line_end_re = regex_atom(f"{whitespace_re}*$")
+repo_sep_re     = regex_atom(f"{sep_comma_re}|{sep_and_re}|{sep_line_end_re}")
```

Comment:
This changes what the current spec says and is not something we want. We do want to allow `, and` as a separator (a lot of people use Oxford comma), and we do want to allow text after the test reference list for potential comments, so we don't want `$`. We also don't really want `<ref> and and and <ref>` (the previous expression allowed multiple commas because I was lazy, but we shouldn't make it worse).

I suggest:
```suggestion
and_re      = regex_atom(f"and ")
ref_sep_re  = regex_atom(f"[, ]+{and_re}?[ ]*")
```